core/include/io/Filter_File_Handle.hpp

```diff
@@ -43,32 +47,45 @@ namespace IO
 		bool Find_in_Line(const std::string & s);
 		// Removes a set of chars from a string
 		void Remove_Chars_From_String(std::string &str, char* charsToRemove);
-
+        // Removes comments from a string
+        bool Remove_Comments_From_String( std::string &str );
+        // Read a string into var
+        void Read_String( std::string& var, const std::string name, bool log_notfound = true );
+        // Count the words of a string
+        int Count_Words( const std::string& str );
+        // get name 
+        
 		// Reads a single variable into var, with optional logging in case of failure.
-		template <typename T> void Read_Single(T & var, const std::string name, bool log_notfound = true)
+		template <typename T> void Read_Single( T & var, const std::string name,  
+                                                bool log_notfound = true )
```

Comment:
Would it be more useful if we had another argument like `bool critical = false` in case that `Read_Single()` cannot find a certain `var`, critical for the data read in? In case that the `var` is not found a `Warning` is logged and the default value is used but is that enough to inform the user for the mishap? Sometimes the default value of `var` might not be meaningful. Same holds for the `Filter_File_Handle::Read_String()`

Reply:
Actually, it might be better design to simply make this function return a bool. The caller can then decide whether to log a not-found or to e.g. `throw`.